novawallet/Modules/AssetList/AssetListInteractor.swift

```diff
@@ -22,18 +22,22 @@ final class AssetListInteractor: AssetListBaseInteractor {
     let settingsManager: SettingsManagerProtocol
     let walletConnect: WalletConnectDelegateInputProtocol
     let assetListModelObservable: AssetListModelObservable
+    let proxyListLocalSubscriptionFactory: ProxyListLocalSubscriptionFactoryProtocol
 
     private var nftSubscription: StreamableProvider<NftModel>?
     private var nftChainIds: Set<ChainModel.Id>?
 
     private var assetLocksSubscriptions: [AccountId: StreamableProvider<AssetLock>] = [:]
     private var locks: [ChainAssetId: [AssetLock]] = [:]
+    private var proxyListSubscription: StreamableProvider<ProxyAccountModel>?
```

Comment:
I would have WalletNotificationService that wraps inside the logic of resolving proxies for current wallet and easy integration to other screens as well as adding new notifications in future